Drivers/NER/src/cascadiamc.c

```diff
@@ -0,0 +1,407 @@
+/**
+ * @file cascadiamc.c
+ * @author Hamza Iqbal
+ * @brief Source file for Motor Controller Driver
+ * @version 0.1
+ * @date 2023-08-22
+ * 
+ * @copyright Copyright (c) 2023
+ * 
+ */
+
+#include "cascadiamc.h"
+
+void cascadiamc_init(cascadiamc_t* mc)
+{
+
+}
+
+void cascadiamc_update(cascadiamc_t* mc, can_msg_t message)
+{
+    switch(message.id)
+    {
+        case 0x2010:    // The byte furthest to the right is a standin for now, should be the last 2 digits of serial number in hex
+            int32_t ERPM = message.data >> 32;
+            mc->rpm = ERPM / NUM_POLES;
+            mc->duty_cycle = ((message.data >> 16) & 0x00000000FFFF);
+            mc->input_voltage = message.data & 0x00000000FFFF;
+        case 0x2110:    // The byte furthest to the right is a standin for now, should be the last 2 digits of serial number in hex
+            mc->ac_current = message.data >> 48;
+            mc->dc_current = ((message.data >> 32) & 0x00000000FFFF);
+        case 0x2210:    // The byte furthest to the right is a standin for now, should be the last 2 digits of serial number in hex
+            mc->cont_temp = message.data >> 48;
+            mc->motor_temp = ((message.data >> 32) & 0x00000000FFFF);
+            mc->fault_code = ((message.data >> 24) & 0x0000000000FF);
+        case 0x2310:    // The byte furthest to the right is a standin for now, should be the last 2 digits of serial number in hex
+            mc->throttle_signal = message.data >> 56;
+            mc->brake_signal = ((message.data >> 48) & 0x0000000000FF);
+            mc->drive_enable = ((message.data >> 39) & 0x000000000001);
+    }
+}
+
+int32_t cascadiamc_get_rpm(cascadiamc_t* mc)
```

Comment:
Tbh I said the getters were good above but like if this is all the function does then we might not need it. It's not like we are returning a private variable or anything. What might be a better framework for "getting" is to have an "updated" field in a struct somewhere that indicates whether the MC callback has changed the data. This can either be done on a single data point level (i.e. has motor controller temp changed?) or on a higher struct level (i.e. has ANY piece of data relating to the motor controller changed?)